core/primitives/src/sharding.rs

```diff
@@ -1302,3 +1263,84 @@ impl EncodedShardChunk {
         }
     }
 }
+
+#[derive(Clone)]
+pub struct EncodedAndShardChunk {
```

Comment:
I would love to get a suggestion for a better name here.

Reply:
gpt suggested `ShardChunkWithEncoding`. Not perfect but works for me.
Also I think it is worth small comment, like "Used to pass chunk around and skip costly encoding/decoding if needed".